server/api/absence-record/absence-record.controller.js

```diff
@@ -198,6 +198,49 @@ exports.chronic = function(req, res) {
   });
 };
 
+function getStudentData(unfilteredArr, stuId) {
+  var total = {};     
+  var entryArr = _.remove(_.map(unfilteredArr, function(entry) {
```

Comment:
This block of code is confusing.

You want to use lodash's built in function, [_.filter](https://lodash.com/docs#filter) instead of using _.remove(_.map(...), undefined)`